File: src/researchmind/models/enums.py
"""Shared enumerations used across the entire ResearchMind pipeline."""

from __future__ import annotations

try:
    from enum import StrEnum
except ImportError:
    # Python < 3.11 fallback
    from enum import Enum

    class StrEnum(str, Enum):
        """Backport of StrEnum for Python < 3.11."""
        pass


class ExtractionRoute(StrEnum):
    """How the PDF was routed through the extraction pipeline."""

    GROBID_PRIMARY = "grobid_primary"
    GROBID_WITH_FALLBACK = "grobid_with_fallback"
    OCR_PRIMARY = "ocr_primary"
    HYBRID = "hybrid"


class ExtractionMethod(StrEnum):
    """Which tool produced a specific piece of extracted content."""

    GROBID = "grobid"
    PYMUPDF_FALLBACK = "pymupdf_fallback"
    OCR_SURYA = "ocr_surya"
    OCR_TESSERACT = "ocr_tesseract"
    LLM_REPAIR = "llm_repair"


class CanonicalLabel(StrEnum):
    """Normalized section labels from the canonical taxonomy."""

    INTRODUCTION = "introduction"
    RELATED_WORK = "related_work"
    METHODOLOGY = "methodology"
    RESULTS = "results"
    DISCUSSION = "discussion"
    CONCLUSION = "conclusion"
    LIMITATIONS = "limitations"
    FUTURE_WORK = "future_work"
    ACKNOWLEDGMENTS = "acknowledgments"
    APPENDIX = "appendix"
    OTHER = "other"


class DocumentType(StrEnum):
    """Classification of the paper's type."""

    RESEARCH_ARTICLE = "research_article"
    REVIEW = "review"
    META_ANALYSIS = "meta_analysis"
    PREPRINT = "preprint"
    THESIS = "thesis"
    CASE_STUDY = "case_study"
    TECHNICAL_REPORT = "technical_report"
    OTHER = "other"


class VenueType(StrEnum):
    """Classification of where the paper was published."""

    JOURNAL = "journal"
    CONFERENCE = "conference"
    WORKSHOP = "workshop"
    PREPRINT_SERVER = "preprint_server"
    THESIS = "thesis"
    UNKNOWN = "unknown"


class ResolutionStatus(StrEnum):
    """Whether a bibliographic reference was successfully resolved."""

    RESOLVED = "resolved"
    UNRESOLVED = "unresolved"
    AMBIGUOUS = "ambiguous"


class ResolutionSource(StrEnum):
    """How a reference was resolved to a canonical identifier."""

    GROBID_CONSOLIDATION = "grobid_consolidation"
    CROSSREF_LOOKUP = "crossref_lookup"
    MANUAL = "manual"
    NONE = "none"


class CitationIntent(StrEnum):
    """Why the author cited a particular reference."""

    SUPPORTS = "supports"
    CONTRASTS = "contrasts"
    EXTENDS = "extends"
    USES_METHOD = "uses_method"
    BACKGROUND = "background"
    COMPARES = "compares"
    UNKNOWN = "unknown"


class ClaimType(StrEnum):
    """Category of a detected candidate claim."""

    STATISTICAL = "statistical"
    CAUSAL = "causal"
    COMPARATIVE = "comparative"
    METHODOLOGICAL = "methodological"
    EXISTENCE = "existence"
    NEGATION = "negation"


class EntityLabel(StrEnum):
    """Named entity type taxonomy."""

    METHOD = "method"
    DATASET = "dataset"
    METRIC = "metric"
    TOOL = "tool"
    MATERIAL = "material"
    DISEASE = "disease"
    DRUG = "drug"
    GENE_PROTEIN = "gene_protein"
    ORGANISM = "organism"
    PERSON = "person"
    ORGANIZATION = "organization"
    LOCATION = "location"
    OTHER = "other"


class StageStatus(StrEnum):
    """Outcome of a pipeline stage execution."""

    SUCCESS = "success"
    PARTIAL = "partial"
    FAILED = "failed"
    SKIPPED = "skipped"

File: frontend/types/enums.ts
export type ExtractionRoute = "grobid_primary" | "grobid_with_fallback" | "ocr_primary" | "hybrid";
export type ExtractionMethod = "grobid" | "pymupdf_fallback" | "ocr_surya" | "ocr_tesseract" | "llm_repair";
export type CanonicalLabel = "introduction" | "related_work" | "methodology" | "results" | "discussion" | "conclusion" | "limitations" | "future_work" | "acknowledgments" | "appendix" | "other";
export type DocumentType = "research_article" | "review" | "meta_analysis" | "preprint" | "thesis" | "case_study" | "technical_report" | "other";
export type VenueType = "journal" | "conference" | "workshop" | "preprint_server" | "thesis" | "unknown";
export type ResolutionStatus = "resolved" | "unresolved" | "ambiguous";
export type ResolutionSource = "grobid_consolidation" | "crossref_lookup" | "manual" | "none";
export type CitationIntent = "supports" | "contrasts" | "extends" | "uses_method" | "background" | "compares" | "unknown";
export type ClaimType = "statistical" | "causal" | "comparative" | "methodological" | "existence" | "negation";
export type EntityLabel = "method" | "dataset" | "metric" | "tool" | "material" | "disease" | "drug" | "gene_protein" | "organism" | "person" | "organization" | "location" | "other";
export type StageStatus = "success" | "partial" | "failed" | "skipped";

export type EvidenceType = "direct_quote" | "paraphrase" | "statistical" | "derived" | "external" | "llm_generated" | "human_annotated";
export type EvidenceTargetType = "claim" | "entity" | "citation_intent" | "confidence";
export type AggregationMethod = "minimum" | "product" | "weighted_average";
export type UncertaintyType = "measurement_error" | "missing_data" | "contradictory_evidence" | "model_uncertainty" | "ambiguous_source" | "extraction_artifact" | "not_applicable";
export type ConsensusStatus = "consistent" | "contradictory" | "insufficient_evidence";

export type ProvenanceAction = "created" | "updated" | "validated" | "corrected" | "rejected" | "annotated" | "merged";
export type ProvenanceAgentType = "pipeline_stage" | "llm" | "human_reviewer" | "external_service" | "rule_based";
export type DataSource = "grobid" | "pymupdf" | "ocr_surya" | "ocr_tesseract" | "crossref" | "pubmed" | "llm" | "manual_review" | "inferred" | "unknown";

export type RelationType = "cites" | "cited_by" | "contradicts" | "supports" | "extends" | "supersedes" | "reproduces" | "uses_method" | "uses_dataset" | "compares_with" | "reviews" | "meta_analysis_includes" | "unknown";
export type AnnotationType = "human_review" | "human_correction" | "llm_suggestion" | "automated_flag" | "quality_issue" | "manual_override";
export type AnnotationStatus = "open" | "accepted" | "rejected" | "superseded";
export type ValidationSeverity = "error" | "warning" | "info";
